CRM/Member/BAO/Membership.php

```diff
@@ -2125,9 +2125,9 @@ public static function setQuickConfigMembershipParameters($membershipOrg, $membe
    */
   protected static function updateDeceasedMembersStatuses() {
     $count = 0;
-    // 'create' context for buildOptions returns only if enabled.
-    $allStatus = self::buildOptions('status_id', 'create');
-    if (($deceasedStatusId = array_search('Deceased', $allStatus)) === FALSE) {
+    // 'match' context for buildOptions returns only if enabled.
+    $allStatus = self::buildOptions('status_id', 'match');
+    if (($deceasedStatusId = array_search('Deceased', array_keys($allStatus))) === FALSE) {
```

Comment:
Just a quick observation (without really thinking about the overall PR/issue). It looks to me like `$deceasedStatusId` comes out with a different value here.

```
$ cv cli
Psy Shell v0.8.0 (PHP 5.6.33 — cli) by Justin Hileman
New version is available (current: v0.8.0, latest: v0.8.17)
>>> CRM_Member_BAO_Membership::buildOptions('status_id', 'create');
=> [
     1 => "New",
     2 => "Current",
     3 => "Grace",
     4 => "Expired",
     5 => "Pending",
     6 => "Cancelled",
     7 => "Deceased",
   ]
>>> array_search('Deceased', CRM_Member_BAO_Membership::buildOptions('status_id', 'create'));
=> 7
>>> array_keys(CRM_Member_BAO_Membership::buildOptions('status_id', 'match'))
=> [
     "New",
     "Current",
     "Grace",
     "Expired",
     "Pending",
     "Cancelled",
     "Deceased",
   ]
>>> array_search('Deceased', array_keys(CRM_Member_BAO_Membership::buildOptions('status_id', 'match')))
=> 6
```

The searches return different numbers because there's no expectation for indexes of `array_keys()` to match IDs in DB.

Reply:
Thanks, I’ve updated my PR: `$deceasedStatusId` is now retrieved using   `CRM_Core_PseudoConstant::getKey()`.